janus_collector/src/lib.rs

```diff
@@ -236,16 +236,75 @@ impl<P> CollectJob<P> {
 
 #[derive(Derivative)]
 #[derivative(Debug)]
-/// The result of a collect request poll operation. This will either provide the aggregate result
+/// The result of a collect request poll operation. This will either provide the collection result
 /// or indicate that the collection is still being processed.
-enum PollResult<T> {
-    /// The aggregate result from a completed collect request.
-    AggregateResult(#[derivative(Debug = "ignore")] T),
+enum PollResult<V: vdaf::Collector>
+where
+    for<'a> Vec<u8>: From<&'a <V as vdaf::Vdaf>::AggregateShare>,
+{
+    /// The collection result from a completed collect request.
+    CollectionResult(#[derivative(Debug = "ignore")] CollectionResult<V>),
     /// The collect request is not yet ready. If present, the [`RetryAfter`] object is the time at
     /// which the leader recommends retrying the request.
     NextAttempt(Option<RetryAfter>),
 }
 
+/// The result of a collection operation.
+#[derive(Debug)]
+pub struct CollectionResult<V: vdaf::Collector>
```

Comment:
I'm not so sure about this name: it's slightly awkward since we sometimes have to handle types like `Result<CollectionResult>`. Suggestions for better names welcome.

Reply:
I suppose it represents a completed collect job, but `CompletedCollectJob` isn't much better than what you've got.